subgraph/src/receipt.ts

```diff
@@ -79,6 +80,22 @@ export function createReceipt(event: ethereum.Event): void {
     log.debug('Creating receipt for base currency lock {}', [lockAddress])
     receipt.payer = event.transaction.from.toHexString()
     receipt.amountTransferred = event.transaction.value
+    // We cannot trust `event.transaction.value` because the purchase function could in fact
+    // be happening inside of a larger transaction whose value is not the amount transfered,
+    // In that case, we need to look up the GNPChanged event
+    if (logs) {
+      for (let i = 0; i < logs.length; i++) {
+        const txLog = logs[i]
+
+        if (
+          // txLog.address == UNLOCK_ADDRESS &&
```

Comment:
I am not sure how to do that since we don't actually have Unlock's address here?